apps/src/javalab/Javalab.js

```diff
@@ -120,7 +139,117 @@ Javalab.prototype.onContinue = function() {
 };
 
 Javalab.prototype.getCode = function() {
-  return '';
+  // store the file version as the source, as we do in WebLab
+  return this.getCurrentFilesVersionId();
+};
+
+Javalab.prototype.getCurrentFilesVersionId = function() {
+  return project.filesVersionId || this.initialFilesVersionId;
+};
+
+// Called by Javalab when a file has been renamed
+Javalab.prototype.renameProjectFile = function(filename, newFilename) {
+  // no need to rename if the name hasn't changed
+  if (filename === newFilename) {
+    return;
+  }
+  header.showProjectSaving();
+  filesApi.renameFile(
+    filename,
+    newFilename,
+    () => {
```

Comment:
JS is not the best at telling a reader when/why an arrow function will be called. Mind adding a comment for these? Guessing it's something like `/* onSuccess */` and `/* onError */`?